From 86eac348aebb18cd9bbf0ec8012bf90132beac88 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Thu, 12 Nov 2015 14:07:48 +0000
Subject: [PATCH] Resolves: tdf#93847 gtk3: paint blocking is unnecessary and
 harmful

for gtk3. Probably also a fairly dumb idea for gtk2, but with the
extra unblocking opportunities it probably just "works out ok" in
that model.

Change-Id: I3d4716c691fc84a5e68c210fee51f9e31cfacc02
(cherry picked from commit f9dfd0129ce0b0e21e3f57488ff513794dbbaae5)
---
 vcl/unx/gtk/window/gtksalframe.cxx | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/vcl/unx/gtk/window/gtksalframe.cxx b/vcl/unx/gtk/window/gtksalframe.cxx
index 980f0e4..13fec40 100644
--- a/vcl/unx/gtk/window/gtksalframe.cxx
+++ b/vcl/unx/gtk/window/gtksalframe.cxx
@@ -947,10 +947,12 @@ void GtkSalFrame::widget_set_size_request(long nWidth, long nHeight)
 {
     gint nOrigwidth, nOrigheight;
     gtk_window_get_size(GTK_WINDOW(m_pWindow), &nOrigwidth, &nOrigheight);
+#if !GTK_CHECK_VERSION(3,0,0)
     if (nWidth > nOrigwidth || nHeight > nOrigheight)
     {
         m_bPaintsBlocked = true;
     }
+#endif
     gtk_widget_set_size_request(m_pWindow, nWidth, nHeight );
 }
 
@@ -958,10 +960,12 @@ void GtkSalFrame::window_resize(long nWidth, long nHeight)
 {
     gint nOrigwidth, nOrigheight;
     gtk_window_get_size(GTK_WINDOW(m_pWindow), &nOrigwidth, &nOrigheight);
+#if !GTK_CHECK_VERSION(3,0,0)
     if (nWidth > nOrigwidth || nHeight > nOrigheight)
     {
         m_bPaintsBlocked = true;
     }
+#endif
     gtk_window_resize(GTK_WINDOW(m_pWindow), nWidth, nHeight);
 }
 
@@ -3658,7 +3662,6 @@ void GtkSalFrame::damaged (const basegfx::B2IBox& rDamageRect)
 gboolean GtkSalFrame::signalDraw( GtkWidget*, cairo_t *cr, gpointer frame )
 {
     GtkSalFrame* pThis = static_cast<GtkSalFrame*>(frame);
-    pThis->m_bPaintsBlocked = false;
 
     cairo_save(cr);
 
-- 
2.5.0

